### PR TITLE
use parser.execute

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
@dnlup I think this will help with the wasm stuff you are working on. Removes usage of `kOnExecute` in favor of `parser.execute`.

**Comment by ronag** [comment]
https://github.com/nodejs/undici/issues/22

File: lib/client-request.js [hunk 1]
```diff
@@ -112,13 +112,14 @@ class RequestHandler extends AsyncResource {
   onComplete (trailers) {
     const { res } = this
 
-    removeSignal(this)
-
-    if (trailers) {
-      util.parseHeaders(trailers, this.trailers)
-    }
+    util.queueMicrotask(() => {
```
Comment (mcollina) [inline lib/client-request.js:115]: why this?

Comment (ronag) [reply lib/client-request.js:115]: `onComplete` can be invoked before `await undici.request()` has continued execution. I think we can avoid the queueMicrotask before landing. For now this just gets tests to pass properly.

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/client-request.js (hunk 1 above)
Comment (ronag) [reply lib/client-request.js]: fixed

**Comment by dnlup** [comment]
> @dnlup I think this will help with the wasm stuff you are working on. Removes usage of `kOnExecute` in favor of `parser.execute`.

Yes, definitely ❤️

**Comment by ronag** [comment]
Has been rebased into https://github.com/nodejs/undici/pull/575

**Pull request closed without merging**